### PR TITLE
MES-4037 -C1 , C1E, CE Vehicle Checks Fault Counters

### PR DESCRIPTION
**Pull request by TomBillingtonUK**
## Description
- Created a new generic method which takes a category for all vehicle checks counters (apart from B) so it matches the other fault counters.
- Added new counters with the correct logic for CE, C1 and C1E

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by amobarak — APPROVED** [review]

File: src/pages/waiting-room-to-car/cat-c/components/vehicle-checks-modal/vehicle-checks-modal.cat-c.page.ts [hunk 1]
```diff
@@ -120,7 +120,7 @@ export class VehicleChecksCatCModal {
         map((vehicleChecks) => {
 
           // TODO: MES-4254 Use cat c Provider
-          return this.faultCountProvider.getVehicleChecksFaultCountCatBE(vehicleChecks);
+          return this.faultCountProvider.getVehicleChecksFaultCount(TestCategory.BE, vehicleChecks);
```
Comment (MarkPatt89) [inline src/pages/waiting-room-to-car/cat-c/components/vehicle-checks-modal/vehicle-checks-modal.cat-c.page.ts:123]: why do you use `BE` here and `C` below? Is the comment still relevant here for future work or can it be done now?

File: src/providers/fault-count/fault-count.ts [hunk 2]
```diff
@@ -122,6 +135,96 @@ export class FaultCountProvider {
     };
   }
 
+  private getVehicleChecksFaultCountCatCE = (vehicleChecks: CatCEUniqueTypes.VehicleChecks): VehicleChecksScore => {
```
Comment (MarkPatt89) [inline src/providers/fault-count/fault-count.ts:138]: These functions could probably be abstracted for readability. There's a bit of reading in them but I'd be happy to approve as the logics sound.

**Review by MarkPatt89 — APPROVED** [review]
Just one comment that may turn out to be nothing around the use of an Enum. I just wasn't sure if the comment above it was still relevant.

File: src/pages/waiting-room-to-car/cat-c/components/vehicle-checks-modal/vehicle-checks-modal.cat-c.page.ts (hunk 1 above)
Comment (TomBillingtonUK) [reply src/pages/waiting-room-to-car/cat-c/components/vehicle-checks-modal/vehicle-checks-modal.cat-c.page.ts:123]: Ahmed is updating this component so his PR will change it :) Nice spot though

File: src/providers/fault-count/fault-count.ts (hunk 2 above)
Comment (TomBillingtonUK) [reply src/providers/fault-count/fault-count.ts:138]: Agreed there is some refactoring work needed on this provider to deal with the duplication/massive file size. I'll take a look at what could be done :)

**Pull request merged**